src/ComposerCompatibleUpdater.php

```diff
@@ -52,7 +52,11 @@ public function getLength(string $target): int
 
         $metadata = $this->getMetadataForTarget($target);
         if ($metadata) {
-            return $metadata->getLength($target);
+            // We need to add 1 to the maximum file size returned by TUF because
+            // if we actually download the expected number of bytes, Composer
+            // will mistakenly think we have exceeded the maximum size, and
+            // throw an exception.
+            return $metadata->getLength($target) + 1;
```

Comment:
Maybe to be clear here we should mention that increasing the size 1 byte does not defeat the purpose of the maximum size that TUF wants us to implement. could we like to part of the protocol that mentions this?
Did you find a composer issue related to this we could link to?

Reply:
I can't find any part of the TUF spec which specifically says that the length check is there to prevent infinite data attacks. But I mentioned it in the comment (just pushed), which is probably good enough.

Reply:
Yes, that's the correct place where the spec mentions the `length` key. What it doesn't say is _why_ the `length` key is there, or what it's used for. That is the relevant info here, IMHO, and I don't think that's mentioned in the spec. That said, it can't hurt to link to the spec either.